File: frontend/recpro-frontend-lib/projects/recpro-frontend-lib/src/lib/recommendation/modeling/recommendation/model/RecommendationServiceModel.ts
import {RecommendationServiceModelType} from './RecommendationServiceModelType';
import {RecommendationServiceModelDimension} from './RecommendationServiceModelDimension';

export class RecommendationServiceModel {
  id: string | undefined;
  name: string = '';
  description: string = '';
  url: string = '';
  recommendationType: RecommendationServiceModelType = RecommendationServiceModelType.BASE;

  dimensions: RecommendationServiceModelDimension[] = [];
}

File: backend/recpro-backend-lib/src/main/java/de/ubt/ai4/petter/recpro/recpro_backend_lib/recpro/model/recommendation/modeling/recommendationServiceModel/RecommendationServiceModel.java
package de.ubt.ai4.petter.recpro.recpro_backend_lib.recpro.model.recommendation.modeling.recommendationServiceModel;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.stereotype.Service;

import java.util.ArrayList;
import java.util.List;
import java.util.UUID;

@Entity
@Data
@Service
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class RecommendationServiceModel {
    @Id
    @Builder.Default
    private String id = UUID.randomUUID().toString();

    private String name;
    private String description;
    private String url;

    @Enumerated(EnumType.STRING)
    @Builder.Default
    private RecommendationServiceModelType recommendationType = RecommendationServiceModelType.BASE;

    @OneToMany(cascade = CascadeType.ALL, orphanRemoval = true)
    @Builder.Default
    private List<RecommendationServiceModelDimension> dimensions = new ArrayList<>();
}
